File: frontend/src/services/questionService.js
import { request } from './apiClient';

export const questionService = {
  getQuestions: (topicId) => request(`/modules/${topicId}/questions`),
  getDeletedQuestions: (topicId) => request(`/modules/${topicId}/questions/deleted`),
  createQuestion: (topicId, data) => request(`/modules/${topicId}/questions`, {
    method: 'POST',
    body: JSON.stringify(data)
  }),
  updateQuestion: (id, data) => request(`/questions/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data)
  }),
  deleteQuestion: (id) => request(`/questions/${id}`, {
    method: 'DELETE'
  }),
  restoreQuestion: (id) => request(`/questions/${id}/restore`, {
    method: 'PUT'
  }),
  reorderQuestions: (orderedIds) => request('/questions/reorder', {
    method: 'PUT',
    body: JSON.stringify(orderedIds)
  })
};

export default questionService;

File: backend/src/main/java/com/coursemanager/service/QuestionService.java
package com.coursemanager.service;

import com.coursemanager.dto.request.QuestionRequest;
import com.coursemanager.dto.response.QuestionResponse;

import java.util.List;

public interface QuestionService {

    List<QuestionResponse> getActiveQuestionsByTopic(Long topicId);

    List<QuestionResponse> getDeletedQuestionsByTopic(Long topicId);

    QuestionResponse getQuestionById(Long id);

    QuestionResponse createQuestion(Long topicId, QuestionRequest request);

    QuestionResponse updateQuestion(Long id, QuestionRequest request);

    void softDeleteQuestion(Long id);

    void restoreQuestion(Long id);

    void reorderQuestions(List<Long> orderedIds);
}
